kafka/integration_test.go

```diff
@@ -395,12 +397,14 @@ func consumerTest(t *testing.T, testname string, assignmentStrategy string, msgc
 
 	}
 
+	if cc.autoCommit {
+		// Sleep for a while to actually let the auto commit happen
+		// We set auto.commit.interval.ms to 500, so give it a bit longer.
+		time.Sleep(1 * time.Second)
+	}
+
 	// Trigger RevokePartitions
 	c.Unsubscribe()
-
-	// Handle RevokePartitions
-	c.Poll(500)
```

Comment:
Why is this removed?

We should be very careful with changing existing tests since any change to tests may hide regression, it is better to add another test-case instead.

Reply:
This does need to go - in the test cases where `go.events.channel.enable` is set, calling `Poll()` here will be called concurrently with `Poll()` in the `consumerReader` loop. The two concurrent calls to Poll() is a bit daft, because it's totally undefined which one will actually receive any messages.

The reason for this being here in the first place, to handle RevokePartitions, is I believe no longer necessary anyway, because `Close()` does this itself now (I think this is a new addition), and does it _after_ shutting down the `consumerReader` loop so that there's only one call to `rd_kafka_queue_poll` in flight at a time.

Reply:
I removed it initially because it caused deadlocks with my first implementation of `rd_kafka_queue_yield`, because that woke up all `rd_kafka_queue_poll` calls I think. I subsequently sure that my implementation could handle concurrent calls to Poll(), but left the test case change here because I don't think concurrent calls to Poll() is something that really _should_ be supported.